Client/mods/deathmatch/logic/CClientModel.h

```diff
@@ -34,6 +34,8 @@ class CClientModel
     eClientModelType                GetModelType ( void ) { return m_eModelType; };
     bool                            Allocate ( void );
     bool                            Deallocate ( void );
+    void                            SetResourceThatAllocatedIt(CResource* pResource) { m_pOwnedByResource = pResource; }
```

Comment:
a better name might be `SetParentResource` with a comment "This is the resource that allocated this model"